scene/gui/texture_progress_bar.cpp

```diff
@@ -510,18 +510,38 @@ void TextureProgressBar::_notification(int p_what) {
 								}
 								pts.append(to);
 
+								AtlasTexture *atlas_progress = Object::cast_to<AtlasTexture>(*progress);
+								bool valid_atlas_progress = atlas_progress && atlas_progress->get_atlas().is_valid();
```

Comment:
This would normally be done with `Ref<T>` instead of going to the raw pointer:
```suggestion
								Ref<AtlasTexture> atlas_progress = progress;
								bool valid_atlas_progress = atlas_progress.is_valid() && atlas_progress.get_atlas().is_valid();
```

Reply:
@akien-mga Thanks for your review!
This is my first time in C++ after ~10 years and I actually don't remember anything, that's why I used this https://github.com/godotengine/godot/blob/84f2c68c84bde10026454292578e5fdcc9ab69a3/scene/2d/cpu_particles_2d.cpp#L164 as reference. 

Pushed changes as you suggested. Please take a look.